packages/gui_setup.rb

```diff
@@ -0,0 +1,40 @@
+require 'package'
+
+class Gui_setup < Package
+  description 'Graphical environment setup for Chromebrew GUI applications'
+  homepage 'https://github.com/skycocker/chromebrew/'
+  version '1.0'
+  license 'GPL-3+'
+  compatibility 'all'
+
+  is_fake
+
+  def self.preflight
+    # exo wayland compositor is required for sommelier
+    abort 'Graphical environment is not supported in your Chrome OS version'.lightred unless File.stat('/var/run/chrome/wayland-0').socket?
+
+    puts <<~EOT
+      Select display server:
+
+      1 = exo: Chrome OS built-in Wayland compositor (Wayland)
+      2 = sommelier: Nested Wayland compositor with X11 and acceleration support (Recommended) (Wayland & X11)
+      0 = Cancel
+
+    EOT
+
+    while @ver = STDIN.gets.chomp
+      case @ver
+      when '0'
+        abort
+      when '1'
+        depends_on 'exo_base'
+        break
+      when '2'
+        depends_on 'sommelier'
+        break
+      else
+        puts "\nPlease select from one of the options or enter 0 to cancel.".yellow
+      end
+    end
+  end
```

Comment:
I'm thinking this entire preflight section should be wrapped in `if ARGV[0] == 'install' ... end`, otherwise upgrade operations will stop and prompt unnecessarily.

Reply:
Actually, upgrade operations should ignore `is_fake` packages.

Reply:
This might be the wrong place to ask, but is there a way to enable a setting here through a command line flag? If this is a dependency, and I'm running a compile for instance, I don't want it to hang waiting for input.

(This is also an issue with hunspell.)

Reply:
We have the same issue with php.  I think it's a crew bug.  We should ignore `is_fake` packages during upgrades.  Maybe we should file an issue in the queue.

Reply:
@satmandu How about setting a timeout for 5 seconds and making `sommelier` as default?

Reply:
I think the main issue here is there is no need to "upgrade" is_fake packages.  The only packages that need upgrades are the dependencies.  The upgrade process should completely ignore is_fake packages because there is never a need to do it.

Reply:
PR #5528 should resolve this issue?  No need to change anything here.